app/models/consultee_constraint.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class ConsulteeConstraint < ApplicationRecord
+  belongs_to :consultee, class_name: "Contact"
+  belongs_to :constraint
+
+  validates :constraint_id, uniqueness: {scope: :consultee_id}
+  validate :consultee_has_consultee_category
+
+  def consultee_has_consultee_category
+    return if consultee&.consultee?
+
+    errors.add(:consultee, :invalid)
+  end
```

Comment:
Took me a moment to understand this validation until I realised that the `consultee` relationship isn't actually to the `consultee` model. I get why, but it feels confusing.

Perhaps if the validation was called something like `contact_is_consultee` or `contact_category_is_consultee` or something?